src/components/web-monitoring-ui.tsx

```diff
@@ -57,3 +71,18 @@ export default class WebMonitoringUi extends React.Component<undefined, IWebMoni
         return {api};
     }
 }
+
+function getPagesByUser (userName: string): Promise<any> {
```

Comment:
You should pull this out into its own module for interacting with the local API (like the `WebMonitoringDb` one for interacting with the DB API).

Also, this should be a `Promise<Page[]>` (same as [`WebMonitoringDb.getPages()`](https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/master/src/services/web-monitoring-db.ts#L63)

Finally, it might be useful to split part of this out into a `getDomainsForUser(user)` function that just returns a promise for an array of domains, so the code reads more like:

```js
function getPagesByUser (userName: string): Promise<Page[]> {
  return getDomainsForUser(user).then(api.getPagesByDomains);
}
```